pkg/controller/infrastructure/actuator_delete.go

```diff
@@ -79,18 +91,28 @@ func Delete(
 		destroyLoadBalancersAndSecurityGroups = g.Add(flow.Task{
 			Name: "Destroying Kubernetes load balancers and security groups",
 			Fn: flow.TaskFn(func(ctx context.Context) error {
-				stateVariables, err := tf.GetStateOutputVariables(ctx, aws.VPCIDKey)
-				if err != nil {
-					if apierrors.IsNotFound(err) || terraformer.IsVariablesNotFoundError(err) {
-						logger.Info("Skipping explicit AWS load balancer and security group deletion because not all variables have been found in the Terraform state.")
-						return nil
+				var vpcID string
+
+				if infrastructureConfig != nil && infrastructureConfig.Networks.VPC.ID != nil {
+					vpcID = *infrastructureConfig.Networks.VPC.ID
+				} else {
+					stateVariables, err := tf.GetStateOutputVariables(ctx, aws.VPCIDKey)
+					if err == nil {
+						vpcID = stateVariables[aws.VPCIDKey]
+					} else if err != nil && !apierrors.IsNotFound(err) && !terraformer.IsVariablesNotFoundError(err) {
```

Comment:
Nit: redundant condition
```suggestion
					} else if !apierrors.IsNotFound(err) && !terraformer.IsVariablesNotFoundError(err) {
```